src/components/CodeChallengeCard.astro

```diff
@@ -0,0 +1,61 @@
+---
+interface Props {
+    title : string,
+    href : string,
+    difficulty : string
+}
+
+const {href, title, difficulty} = Astro.props;
```

Comment:
You declared a props interface but I don't think you used it. Should the interface annotate this declaration?

Reply:
As i understand it, the props interface tells astro that i will give the code challenge component these parameters, and then in the layout part, i am using this parameters

Reply:
![Screenshot 2024-10-23 at 16 39 35](https://github.com/user-attachments/assets/3940a369-50db-499b-959f-0e47b755dd65)

used here

Reply:
That part is fine. But where is the type safety if you declare an interface without using it?

Reply:
In my mind, line 8 should be:
```
const {href, title, difficulty}: Props = Astro.props;
```
It looks like we have the same problem in the Layout file, but I only just noticed it

Reply:
Although in my editor, when I look at the Layout props, it somehow knows that `title` is a `string`.
Will resolve this comment for now as it's my understanding that's at fault, I think